guillotina/transactions.py

```diff
@@ -131,24 +131,27 @@ async def __aexit__(self, exc_type, exc, tb):
                 self.txn.deleted = {**self.previous_txn.deleted, **self.txn.deleted}
                 self.txn.added = {**self.previous_txn.added, **self.txn.added}
 
-        if self.abort_when_done:
+        if exc:
             await self.tm.abort(txn=self.txn)
         else:
-            await self.tm.commit(txn=self.txn)
-
-        if self.adopt_parent_txn and self.previous_txn is not None:
-            # restore transaction ownership of item from adoption done above
-            if self.previous_txn != self.txn:
-                # we adopted previously detetected transaction so now
-                # we need to clear changed objects and restore ownership
-                self.previous_txn.modified = {}
-                self.previous_txn.deleted = {}
-                self.previous_txn.added = {}
-
-        if self.execute_futures:
-            from guillotina.utils import execute
-
-            execute.execute_futures()
+            if self.abort_when_done:
+                await self.tm.abort(txn=self.txn)
+            else:
+                await self.tm.commit(txn=self.txn)
+
+            if self.adopt_parent_txn and self.previous_txn is not None:
```

Comment:
I think we should clear the registered changed objects no matter what. Do you agree?

Reply:
I wasn't sure but it's ok :)